### PR TITLE
New package: vault

### PR DESCRIPTION
**Pull request by avnik**

### CODE REVIEW

File: pkgs/servers/vault/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    '';
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/vault $out/bin/vault
+    '';
+
+    meta = with stdenv.lib; {
+      homepage = "https://www.vaultproject.io";
+      description = ''
```
Comment (jagajaga) [inline pkgs/servers/vault/default.nix]: Read how to write decription https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

Comment (avnik) [inline pkgs/servers/vault/default.nix]: On Thu, Jun 25, 2015 at 02:58:24PM -0700, Arseniy Seroka wrote:

> > +    meta = with stdenv.lib; {
> > +      homepage = "https://www.vaultproject.io";
> > +      description = ''
> 
> Read how to write decription https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes
> 

Fixed.
Shorten to one-liner.

File: pkgs/servers/vault/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bin/vault $out/bin/vault
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://www.vaultproject.io";
+    description = "A tool for securely accessing secrets.";
```
Comment (jagajaga) [inline pkgs/servers/vault/default.nix]: Remove dot at the end :)

Comment (avnik) [inline pkgs/servers/vault/default.nix]: On Thu, Jun 25, 2015 at 03:33:05PM -0700, Arseniy Seroka wrote:

> > +  meta = with stdenv.lib; {
> > +    homepage = "https://www.vaultproject.io";
> > +    description = "A tool for securely accessing secrets.";
> 
> Remove dot at the end :)

done

**Comment by wkennington** [comment]
We already have vault in gopackages.

**Comment by jagajaga** [comment]
Oooooops.

**Comment by wkennington** [comment]
This also isn't using buildGoPackage so I recommend reverting.

**Comment by jagajaga** [comment]
@wkennington can you revert please?

**Comment by avnik** [comment]
On Thu, Jun 25, 2015 at 04:00:03PM -0700, William A. Kennington III wrote:

I am a bit worrying about build against random versions of dependencies.
Also I am curious, do we have some tool to detect build inputs, which already
packaged?

> This also isn't using buildGoPackage so I recommend reverting.
> 
> > Oooooops.
> >

**Comment by wkennington** [comment]
It should be fine to use up to date dependencies as long as everything builds and the tests pass. Pinning dependencies has the issue of sitting on old and potentially vulnerable versions of software, when we could otherwise have an automated vetting process.

**Comment by avnik** [comment]
On Thu, Jun 25, 2015 at 11:25:42PM -0700, William A. Kennington III wrote:

> It should be fine to use up to date dependencies as long as everything
> builds and the tests pass. Pinning dependencies has the issue of sitting
> on old and potentially vulnerable versions of software, when we could
> otherwise have an automated vetting process.
> 

few more questions:

1 How I can build <https://github.com/hashicorp/terraform> which provide more
than one binary? (they use gox to determine what paths need to be built)

2 Should we propagate `vault` (and `terraform`) from goPackages to pkgs
(like consul does)

3 Should we add meta to go packages.

4 How I can query packages, already added in goPackages with their `import path`

**Comment by copumpkin** [comment]
@avnik if you look at my `hologram` package, I think you'll find that it produces more than one binary in the `goPackages` framework.

**Comment by avnik** [comment]
On Fri, Jun 26, 2015 at 07:01:26AM -0700, Daniel Peebles wrote:

> @avnik if you look at my `hologram` package, I think you'll find that it produces more than one binary in the `goPackages` framework.
> 

@copumpkin I haven't see anything special in `hologram`.
<https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/go-packages.nix#L1567>

**Comment by copumpkin** [comment]
@avnik that's my point, it doesn't do anything special but it produces multiple binaries :smile:

**Pull request merged**